diff --git a/wmd-editor/pagedown/Markdown.Editor.js b/wmd-editor/pagedown/Markdown.Editor.js
index e3205cc..cac0cad 100644
--- a/wmd-editor/pagedown/Markdown.Editor.js
+++ b/wmd-editor/pagedown/Markdown.Editor.js
@@ -111,6 +111,10 @@
                                                   * its own image insertion dialog, this hook should return true, and the callback should be called with the chosen
                                                   * image url (or null if the user cancelled). If this hook returns false, the default dialog will be used.
                                                   */
+        hooks.addFalse("insertLinkDialog");      /* called with one parameter: a callback to be called with the URL of the link. If the application creates
+                                                  * its own image insertion dialog, this hook should return true, and the callback should be called with the chosen
+                                                  * image url (or null if the user cancelled). If this hook returns false, the default dialog will be used.
+                                                  */
 
         this.getConverter = function () { return markdownConverter; }
 
@@ -1811,7 +1815,8 @@
                     ui.prompt(this.getString("imagedialog"), imageDefaultText, linkEnteredCallback);
             }
             else {
-                ui.prompt(this.getString("linkdialog"), linkDefaultText, linkEnteredCallback);
+                if (!this.hooks.insertLinkDialog(linkEnteredCallback))
+                    ui.prompt(this.getString("linkdialog"), linkDefaultText, linkEnteredCallback);
             }
             return true;
         }
@@ -2240,4 +2245,4 @@
     }
 
 
-})();
\ No newline at end of file
+})();
